js/src/Ice/ConnectionI.js

```diff
@@ -1077,12 +1074,11 @@ export class ConnectionI {
         os.writeByte(0); // compression status: always report 0 for CloseConnection.
         os.writeInt(Protocol.headerSize); // Message size.
 
-        if ((this.sendMessage(OutgoingMessage.createForStream(os, false)) & AsyncStatus.Sent) > 0) {
-            //
-            // Schedule the close timeout to wait for the peer to close the connection.
-            //
-            this.scheduleTimeout(SocketOperation.Read);
+        if (this._closeTimeout > 0) {
```

Comment:
In other languages, we also schedule this CloseTimeout when we receive a CLOSE_CONNECTION message from the peer, because "closing" can occasionally take a while.

It's not immediately clear to me why this is different in JS.

Reply:
I looked a bit at this. I think it is because WS transport needs to handle the close connection at the WS level, for TCP the transceiver returns immediately when calling closing on a client connection.

https://github.com/zeroc-ice/ice/blob/61662730fbe8845c8c9641a828585c4caf484a88/csharp/src/Ice/ConnectionI.cs#L2267-L2271

https://github.com/zeroc-ice/ice/blob/61662730fbe8845c8c9641a828585c4caf484a88/csharp/src/Ice/Internal/TcpTransceiver.cs#L21-L25

If you look at the Ice protocol close connection description it states that the client can close the read and write ends of the connection once it receives a close connection message from the server. https://doc.zeroc.com/ice/3.7/ice-protocol-and-encoding/protocol-messages#ProtocolMessages-CloseConnectionMessage